subsys/bluetooth/host/gatt.c

```diff
@@ -657,6 +683,15 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 	/* Update cfg if don't match */
 	if (ccc->cfg[i].value != ccc->value) {
 		gatt_ccc_changed(attr, ccc);
+
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+		if (atomic_cas(&gatt_ccc_store.scheduled, false, true) &&
+		    bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+			gatt_ccc_store.conn = conn;
```

Comment:
This one I didn't notice earlier: bt_conn pointers should always be copied with bt_conn_ref() so that the reference counting works as expected. I.e. the above should be `gatt_ccc_store.conn = bt_conn_ref(conn);` and then you'll need to add the corresponding bt_conn_unref() in the right place(s).

Reply:
Fixed